sdk/cargo-build-bpf/src/main.rs

```diff
@@ -247,6 +247,71 @@ fn postprocess_dump(program_dump: &Path) {
     fs::rename(postprocessed_dump, program_dump).unwrap();
 }
 
+// Check whether the built .so file contains undefined symbols that are
+// not known to the runtime and warn about them if any.
+fn check_undefined_symbols(config: &Config, program: &Path) {
```

Comment:
Any way we can pull these symbols from the BPF SDK, or somewhere under sdk/?  We'll 100% forget to update this list the next time a new symbol is added

Reply:
could parse the syscalls.rs source file in build.rs maybe when cargo-build-bpf is built, but seems not trivial.

Reply:
Yeah that definitely doesn't seem great either.  I guess this is just a warning, so not too bad when we forget to add a new symbol in here

Reply:
I agree, adding a warning about unresolved symbols will cause confusion and maybe delay/block a developer from doing something that is actually ok.  100% these lists are going to get out of sync and worse folks are already using the build tools across networks and feature activation.  There is no knowledge at this point in the sdk about available syscalls and that could also get out of sync with the network.

I don't have a good idea on how to solve this issue.  Maybe some way to query the network to get compatibility info?